spec.md

```diff
@@ -0,0 +1,954 @@
+  * A Plugin Supervisor MAY programmatically evaluate or otherwise scan a Plugin Package’s image manifest in order to discover configurable environment variables.
+  * A Plugin SHALL NOT assume that an operator or Plugin Supervisor will scan an image manifest for environment variables.
+
+#### Environment Variables
+
+* Variables defined by this specification SHALL be identifiable by their `CMI_` name prefix.
+* Configuration properties not defined by the CMI specification SHALL NOT use the same `CMI_` name prefix; this prefix is reserved for common configuration properties defined by the CMI specification.
+* The Plugin Supervisor SHOULD supply all RECOMMENDED CMI environment variables to a Plugin.
+* The Plugin Supervisor SHALL supply all REQUIRED CMI environment variables to a Plugin.
+
+##### `CMI_ENDPOINT`
+
+Network endpoint at which a Plugin SHALL host CMI RPC services. The general format is:
+
+    {scheme}://{authority}{endpoint}
+
+The following address types SHALL be supported by Plugins:
+
+    unix:///path/to/unix/socket.sock
+
+Note: All UNIX endpoints SHALL end with `.sock`. See [gRPC Name Resolution](https://github.com/grpc/grpc/blob/master/doc/naming.md).
+
+This variable is REQUIRED.
+
+#### Operational Recommendations
+
+The Plugin Supervisor expects that a Plugin SHALL act as a long-running service vs. an on-demand, CLI-driven process.
+
+Supervised plugins MAY be isolated and/or resource-bounded.
+
+##### Logging
+
+* Plugins SHOULD generate log messages to ONLY standard output and/or standard error.
+  * In this case the Plugin Supervisor SHALL assume responsibility for all log lifecycle management.
+* Plugin implementations that deviate from the above recommendation SHALL clearly and unambiguously document the following:
+  * Logging configuration flags and/or variables, including working sample configurations.
+  * Default log destination(s) (where do the logs go if no configuration is specified?)
+  * Log lifecycle management ownership and related guidance (size limits, rate limits, rolling, archiving, expunging, etc.) applicable to the logging mechanism embedded within the Plugin.
+* Plugins SHOULD NOT write potentially sensitive data to logs (e.g. secrets).
+
+##### Available Services
+
+* Plugin Packages MAY support all or a subset of CMI services; service combinations MAY be configurable at runtime by the Plugin Supervisor.
+  * A plugin MUST know the "mode" in which it is operating (e.g. node, controller, or both).
+  * This specification does not dictate the mechanism by which mode of operation MUST be discovered, and instead places that burden upon the VM Provider.
+* Misconfigured plugin software SHOULD fail-fast with an OS-appropriate error code.
+
+##### Linux Capabilities
+
+* Plugin Supervisor SHALL guarantee that plugins will have `CAP_SYS_ADMIN` capability on Linux when running on Nodes.
+* Plugins SHOULD clearly document any additionally required capabilities and/or security context.
+
+##### Namespaces
```

Comment:
Again, this is applicable to CSI, and not CMI

Reply:
You are right this is not relevant. Hence removing this